browser/brave_app_controller_mac.mm

```diff
@@ -0,0 +1,103 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#import "brave/browser/brave_app_controller_mac.h"
+
+#include "brave/app/brave_command_ids.h"
+#include "brave/browser/ui/browser_commands.h"
+#include "chrome/app/chrome_command_ids.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_commands.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/grit/generated_resources.h"
+
+@implementation BraveAppController
+
+- (void)mainMenuCreated {
+  [super mainMenuCreated];
+
+  NSMenu* editMenu = [[[NSApp mainMenu] itemWithTag:IDC_EDIT_MENU] submenu];
+  _copyMenuItem = [editMenu itemWithTag:IDC_CONTENT_CONTEXT_COPY];
+  DCHECK(_copyMenuItem);
+  [[_copyMenuItem menu] setDelegate:self];
+  _copyCleanLinkMenuItem = [editMenu itemWithTag:IDC_COPY_CLEAN_LINK];
+  DCHECK(_copyCleanLinkMenuItem);
+  [[_copyCleanLinkMenuItem menu] setDelegate:self];
+}
+
+- (void)dealloc {
+  [[_copyMenuItem menu] setDelegate:nil];
+  [[_copyCleanLinkMenuItem menu] setDelegate:nil];
+  [super dealloc];
+}
+
+- (Browser*)getBrowser {
+  return chrome::FindBrowserWithProfile([self lastProfileIfLoaded]);
+}
+
+- (BOOL)shouldShowCleanLinkItem {
+  if (_hasSelectedURLForTesting.has_value()) {
+    return _hasSelectedURLForTesting.value();
+  }
+  return brave::HasSelectedURL([self getBrowser]);
+}
+
+- (void)setKeyEquivalentToItem:(NSMenuItem*)item {
+  auto* hootkeyItem =
+      item == _copyMenuItem ? _copyMenuItem : _copyCleanLinkMenuItem;
+  auto* noHootkeyItem =
+      item == _copyMenuItem ? _copyCleanLinkMenuItem : _copyMenuItem;
+
+  [hootkeyItem setKeyEquivalent:@"c"];
```

Comment:
this PR should not do anything to change the shortcut, it should only be adding the menu item

Reply:
removed